web/src/settings_account.ts

```diff
@@ -272,6 +273,32 @@ export function hide_confirm_email_banner(): void {
     $("#account-settings-status").hide();
 }
 
+export function update_user_self_role_select_state(): void {
+    const $select_elem = $("select#user-self-role-select");
+    if ($select_elem.has("data-tippy-content")) {
+        ui_util.enable_element_and_remove_tooltip($select_elem);
+    }
+    if (!people.user_can_change_their_own_role()) {
+        $select_elem.attr("disabled", "true");
+        if (people.is_current_user_only_owner() && current_user.is_owner) {
+            ui_util.disable_element_and_add_tooltip(
+                $select_elem,
+                "The owner role cannot be removed from the only organization owner.",
```

Comment:
Tooltip is not shown initially because this function is only called from the live-update code.